hrm-domain/scheduled-tasks/conversation-cleanup/index.ts

```diff
@@ -0,0 +1,234 @@
+        state ? `&State=${state}` : ''
+      }`,
+      {
+        headers: {
+          Authorization: `Basic ${Buffer.from(`${accountSid}:${authToken}`).toString(
+            'base64',
+          )}`,
+        },
+      },
+    );
+    if (!response.ok) {
+      throw new Error(
+        `Error retrieving conversations from Twilio API for URL ${response.url} Status: ${
+          response.status
+        }: ${await response.text()}`,
+      );
+    }
+    const body = (await response.json()) as ConversationListResponse;
+    const { conversations } = body;
+    conversationsRetrieved = conversations.length;
+    const conversationSids = conversations.map(c => c.sid);
+    if (conversationSids.length > 0) {
+      if (errorIfFound) {
+        console.error(
+          `POSSIBLE STALE CONVERSATIONS DETECTED! ${
+            conversationSids.length
+          } conversations found older than ${retentionDays}${
+            state ? ` in a '${state}' state` : ''
+          } for account ${accountSid}:`,
+          conversationSids,
+        );
+      }
+      await removeConversationsBySids(
+        accountSid,
+        conversationSids,
+        state && state !== 'closed',
+      );
+    }
+    // This prevents us exceeding the twilio imposed rate limit
+    const duration = Date.now() - start;
+    if (duration < ITERATION_DELAY) {
+      await delay(ITERATION_DELAY - duration);
+    }
+  }
+  if (iteration >= MAX_ITERATIONS) {
+    console.warn(
+      `Repeated the pull for 100 conversations for cleanup for account ${accountSid} ${iteration} ${
+        state ? `(state '${state}')` : ''
+      } times, this could indicate an issue with the cleanup logic`,
+    );
+  }
+};
+
+/**
+ * Cleanup all pending cleanup jobs
+ * @returns void
+ * @throws Error
+ */
+export const cleanupConversations = async (): Promise<void> => {
+  await loadSsmCache({
+    configs: [
+      {
+        path: `/${process.env.NODE_ENV}/twilio`,
+        regex: /\/.*\/account_sid/,
+      },
+      {
+        path: `/${process.env.NODE_ENV}/twilio`,
+        regex: /\/.*\/auth_token/,
+      },
+      {
+        path: `/${process.env.NODE_ENV}/hrm`,
+        regex: /\/.*\/transcript_retention_days/,
+      },
+    ],
+    cacheDurationMilliseconds: 1000 * 60 * 60, // 1 hour
+  });
+
+  const accountSids: AccountSID[] = Object.values(
+    getCachedParameters(/^\/[^\/]+\/twilio\/[^\/]+\/account_sid$/),
+  ) as AccountSID[];
+
+  console.info(`Cleaning up contact jobs for accounts:`, accountSids);
+
+  for (const accountSid of accountSids) {
+    console.info(`Checking for possible stuck conversations for account:`, accountSid);
+    await removeOldConversations(
+      accountSid,
+      MAX_ACTIVE_CONVERSATION_HOURS,
+      true,
+      'active',
+    );
+    await removeOldConversations(
+      accountSid,
+      MAX_ACTIVE_CONVERSATION_HOURS,
+      true,
+      'inactive',
+    );
```

Comment:
Does this forcefully removes ALL conversations for accounts even if `inactive`, every 24 hours? 🤔